plugins/woocommerce/client/admin/client/core-profiler/components/geolocation-country-select/geolocation-country-select.tsx

```diff
@@ -0,0 +1,157 @@
+/**
+ * External dependencies
+ */
+import {
+	useEffect,
+	useState,
+	createInterpolateElement,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button, Notice } from '@wordpress/components';
+import { SelectControl } from '@woocommerce/components';
+import { Icon, chevronDown } from '@wordpress/icons';
+import { findCountryOption, getCountry } from '@woocommerce/onboarding';
+import { GeolocationResponse } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import type { CountryStateOption } from '../../services/country';
+
+type Props = {
+	countries: CountryStateOption[];
+	geolocatedLocation?: GeolocationResponse | null;
+	initialValue: CountryStateOption;
+	label: string;
+	placeholder: string;
+	onChange: ( country: CountryStateOption ) => void;
+	onGeolocationOverruledChange?: ( overruled: boolean ) => void;
+};
+
+export const GeolocationCountrySelect = ( {
+	countries,
+	geolocatedLocation,
+	initialValue,
+	label,
+	placeholder,
+	onChange,
+	onGeolocationOverruledChange,
+}: Props ) => {
+	const [ selectedCountry, setSelectedCountry ] =
+		useState< CountryStateOption >( initialValue );
+
+	const [ geolocationMatch, setGeolocationMatch ] =
+		useState< CountryStateOption >( { key: '', label: '' } );
+
+	const [ dismissedNotice, setDismissedNotice ] = useState( false );
+
+	useEffect( () => {
+		setSelectedCountry( initialValue );
+	}, [ initialValue ] );
+
+	useEffect( () => {
+		if ( geolocatedLocation ) {
+			const match = findCountryOption( countries, geolocatedLocation );
+			if ( match ) {
+				setGeolocationMatch( match );
+				if ( ! initialValue?.key ) {
+					setSelectedCountry( match );
+					onChange( match );
+				}
+			}
+		}
+	}, [ countries, geolocatedLocation, initialValue?.key ] );
+
+	const [ geolocationOverruled, setGeolocationOverruled ] = useState( false );
+
+	useEffect( () => {
+		const overruled = Boolean(
+			geolocatedLocation &&
+				getCountry( selectedCountry?.key ) !==
+					getCountry( geolocationMatch?.key )
+		);
+
+		setGeolocationOverruled( overruled );
+		onGeolocationOverruledChange?.( overruled );
+	}, [ selectedCountry, geolocationMatch, geolocatedLocation ] );
+
+	return (
+		<div className="woocommerce-geolocation-country-select">
+			<SelectControl
+				className="woocommerce-profiler-select-control__country"
+				instanceId={ 2 }
```

Comment:
Would something more identifiable be better?

Reply:
@psealock If we're referring to instanceId, it unfortunately only accepts a number. See: https://github.com/WordPress/gutenberg/blob/bdb16796eb85051b9d5d7b00fcb773973642a5e7/packages/compose/src/higher-order/with-instance-id/index.tsx#L11